src/ui/Distance/DistanceResources.ts

```diff
@@ -244,47 +246,42 @@ export class DistanceResources extends Component {
     return this.lastPositionRequest || Promise.reject('No position request was executed yet.');
   }
 
-  private executeQuery() {
-    // Since this DistanceResource component often blocks initial queries,
-    // and relaunch the query automatically when it is able to do so (when a position provider resolves)
-    // we need to have a mechanism to still log something useful for usage analytics, instead of always a "position set".
-    // We want it to be "interface load" on a direct page access, or a "search from link" if coming from an external standalone search box
-    // Everytime this component blocks a query, we keep a copy of the pending search event, and resend this instead of a generic "position set" event.
-    if (this.pendingSearchEventOnCancellation) {
-      this.usageAnalytics.logSearchEvent(this.getPendingEventOnCancellation(), this.pendingSearchEventOnCancellation.getEventMeta());
-      delete this.pendingSearchEventOnCancellation;
-    } else {
-      this.usageAnalytics.logSearchEvent(analyticsActionCauseList.positionSet, {});
-    }
-    this.queryController.executeQuery();
+  private sendAnalytics() {
+    this.usageAnalytics.logSearchEvent(analyticsActionCauseList.positionSet, {});
   }
 
   private shouldTriggerQueryWhenPositionSet() {
-    // If query has been cancelled, we need to trigger the first one.
-    const triggerFirstQueryWithPosition = this.options.cancelQueryUntilPositionResolved && !this.isFirstPositionResolved;
-    return this.options.triggerNewQueryOnNewPosition || triggerFirstQueryWithPosition;
+    // Don't trigger the query if the first one is not yet executed.
+    return !this.queryController.firstQuery && this.options.triggerNewQueryOnNewPosition;
```

Comment:
Just to make sure, are we considering the option :  `this.options.cancelQueryUntilPositionResolved` ? Should we consider it?

Reply:
The option is still used, just not in the same way. The result is pretty much the same for the end-user.

Here: https://github.com/coveo/search-ui/blob/a532c9beceab81a039845a2cf1d16dd7debeff23/src/ui/Distance/DistanceResources.ts#L267-L269

It could be renamed to `deferInitializationUntilPositionResolved` though 🤔